src/python/tests/manifest/test_manifest.py

```diff
@@ -0,0 +1,219 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `ensembl.io.genomio.manifest.manifest` module."""
+
+from contextlib import nullcontext as no_raise
+import json
+from pathlib import Path
+from shutil import copytree
+from typing import Callable, ContextManager
+
+import pytest
+from pytest import LogCaptureFixture, param, raises
+
+from ensembl.io.genomio.manifest.manifest import Manifest, ManifestError
+
+
+_CONTENT_MD5 = "45685e95985e20822fb2538a522a5ccf"
+
+
+@pytest.mark.dependency()
+def test_init(tmp_path: Path) -> None:
+    """Tests `Manifest.__init__()`."""
+    _ = Manifest(tmp_path)
+
+
+@pytest.mark.parametrize(
+    "file_name, expected_name",
+    [
+        param("gene_models.gff3", "gff3", id="Recognised name and suffix"),
+        param("foobar.gff3", "gff3", id="Recognised suffix"),
+        param("genome.json", "genome", id="Recognised name"),
+        param("foobar_seq_region.json", "seq_region", id="Recognised part of the name"),
+        param("foobar.json", "", id="Not recognized json"),
+    ],
+)
+@pytest.mark.dependency(depends=["test_init"])
+def test_get_files_checksum(tmp_path: Path, file_name: str, expected_name: str) -> None:
+    """Tests `Manifest.get_files_checksum()`.
+
+    Fixtures: tmp_path
+
+    Args:
+        tmp_path: Test tmp dir.
+        expected_name: Manifest key expected.
+        file_name: File to create for the test.
```

Comment:
```suggestion
        file_name: File to create for the test.
        expected_name: Manifest key expected.
```
To preserve same order as in the function definition.